src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,96 @@
+
+          setSelectedUser(user);
+        } catch {
+          setError('Something went wrong!');
+        }
+      } else {
+        setSelectedUser(null);
+      }
+    };
+
+    loadData();
+  }, [openedTodo]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
-
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
-
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
-
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
-
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
-        </div>
-      </div>
+      {!selectedUser && !error
+        ? (
+          <Loader />
+        ) : (
+          <>
+            {error && (
+              <p className="notification is-warning">
+                Something went wrong!
+              </p>
+            )}
+
+            {openedTodo && selectedUser && (
+              <div className="modal-card">
+                <header className="modal-card-head">
+                  <div
+                    className="modal-card-title has-text-weight-medium"
+                    data-cy="modal-header"
+                  >
+                    {`Todo #${openedTodo.id}`}
+                  </div>
+
+                  <button
+                    type="button"
+                    className="delete"
+                    data-cy="modal-close"
+                    aria-label="modal-close"
+                    onClick={() => remove()}
+                  />
+                </header>
+
+                <div className="modal-card-body">
+                  <p className="block" data-cy="modal-title">
+                    {openedTodo.title}
+                  </p>
+
+                  <p className="block" data-cy="modal-user">
+                    {
+                      openedTodo.completed
+                        ? <strong className="has-text-success">Done</strong>
+                        : (
+                          <strong className="has-text-danger">
+                            Planned
+                          </strong>
+                        )
+                    }
+
+                    {' by '}
+
+                    <a href={`mailto:${selectedUser.email}`}>
+                      {selectedUser.name}
+                    </a>
```

Comment:
please add check for `selectedUser`
```suggestion
                    {selectedUser && (
                      <a href={`mailto:${selectedUser.email}`}>
                        {selectedUser.name}
                      </a>
                    )}
```